iot_config/src/gateway_service.rs

```diff
@@ -71,7 +72,12 @@ impl iot_config::Gateway for GatewayService {
             .await
             .map_err(|_| Status::internal("gateway lookup error"))?;
 
-        let location = location.ok_or_else(|| Status::internal("gateway location undefined"))?;
+        let location = {
+            let location =
+                location.ok_or_else(|| Status::internal("gateway location undefined"))?;
+            Cell::from_raw(location)
```

Comment:
I felt like adding the is-valid-cell check to this module made sense, thus keeping invalid indices out of the `get_region` call.